cardano-cli/src/Cardano/CLI/Shelley/Run/Query.hs

```diff
@@ -380,6 +490,132 @@ writeLedgerState mOutFile qState@(SerialisedLedgerState serLedgerState) =
      first (const ls) (decodeFull ls)
 
 
+-- Data Structure to hold stake information so that it can be printed nicely
+data Stakes =  Stakes {
+      mark, set, go, total :: Integer
+    } deriving Show
+
+instance ToJSON Stakes where
+    -- this generates a Value
+    toJSON (Stakes m s g t) =
+        object ["activeStakeMark" .= m, "activeStakeSet" .= s, "activeStakeGo" .= g, "activeStakeTotal" .= t]
+
+    -- this encodes directly to a bytestring Builder
+    toEncoding (Stakes m s g t) =
+        pairs ("mark" .= m <> "set" .= s  <> "go" .= g  <> "total" .= t)
+
+
+writeStakeSnapshot :: forall era ledgerera.
+                    ShelleyLedgerEra era ~ ledgerera
+                 => Era ledgerera
+                 => FromCBOR (LedgerState era)
+                 =>
+                    SerialisedLedgerState era
+                 -> Hash StakePoolKey
+                 -> ExceptT ShelleyQueryCmdError IO ()
+
+writeStakeSnapshot qState poolId =
+       case decodeLedgerState qState of
+           Left bs ->
+                      firstExceptT ShelleyQueryCmdHelpersError $ pPrintCBOR bs
+           Right ledgerState ->
+                 if maybehk == Nothing then
+                     left ShelleyQueryCmdPoolIdError
+                 else
+                     liftIO . LBS.putStrLn $  encodePretty $ Stakes {mark = markStake,set = setStake,go = goStake,total = totalStake}
+                 where
+                       -- Ledger State
+                       (LedgerState snapshot)  = ledgerState
+
+                       -- The three stake snapshots
+                       (SnapShots markS setS goS _) = esSnapshots $ nesEs snapshot
+
+                       -- calculation of total stake - this assumes a fixed supply - seems hard to get the variable value
+                       maxSupply = Coin $ 45 * 1000 * 1000 * 1000 * 1000 * 1000
+                       Coin totalStake = circulation (nesEs snapshot) maxSupply
+
+                       -- Calculate the three stake values for the given pool
+                       markStake =   getPoolStake hk markS
+                       setStake =    getPoolStake  hk setS
+                       goStake =     getPoolStake  hk goS
+
+                       -- Convert the hash string into a KeyHash for use by the ledger
+                       maybehk = hashFromStringAsHex $ filter (/= '"') $ show poolId
+                       hk = KeyHash $ fromJust $ maybehk
+
+ where
+   decodeLedgerState
+     :: SerialisedLedgerState era
+     -> Either LBS.ByteString (LedgerState era)
+   decodeLedgerState (SerialisedLedgerState (Serialised ls)) =
+     first (const ls) (decodeFull ls)
+
+--   getPoolStake
```

Comment:
Why is this type signature commented out?

Reply:
> 1. I would like to request a change to the pool-params output to make it more clear what the three different sets returned mean. See suggestion below.
> 2. In addition to this, I would like for the future set an additional field called `activeInEpoch` containing what epoch the future pool update becomes active.
> 
> **Current output:**
> An array containing three unnamed sets for current, future, and retired.
> 
> ```
> [
>   {
>     "publicKey": "0b08b536ba3afe21fbd826fb92bee897a3d1e575a7849b2e6c4659ab",
>     "...": "rest of the fields"
>   },
>   {
>     "publicKey": "0b08b536ba3afe21fbd826fb92bee897a3d1e575a7849b2e6c4659ab",
>     "...": "rest of the fields"
>   },
>   null
> ]
> ```
> 
> **Requested output:**
> My idea would be to have a schema like this instead, replacing the JSON array containing three unnamed objects with a key/value schema where `active` and `future` contains a JSON object(null for future if there are none) and `retired` the epoch number or null.
> Makes it clean and easy to understand what is what.
> 
> ```
> {
>   "active": {
>     "publicKey": "0b08b536ba3afe21fbd826fb92bee897a3d1e575a7849b2e6c4659ab",
>     "...": "rest of the fields"
>   },
>   "future": {
>     "publicKey": "0b08b536ba3afe21fbd826fb92bee897a3d1e575a7849b2e6c4659ab",
>     "activeInEpoch": 125,
>     "...": "rest of the fields"
>   },
>   "retired": null
> }
> ```

Done